<!DOCTYPE html>
<!-- saved from url=(0045)https://andreasviklund.com/files/demo/leaves/ -->
<html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
	
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-155064315-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-155064315-1');
</script>

	<meta name="description" content="Calvin McCarter">
	<meta name="author" content="Calvin McCarter">
	<link href="./css_files/css" rel="stylesheet" type="text/css">
	<link href="./css_files/css(1)" rel="stylesheet" type="text/css">
	<link rel="stylesheet" type="text/css" href="./css_files/leaves.css" title="Leaves" media="all">
	<meta name="viewport" content="width=device-width, initial-scale=0.6">
	<title>Calvin McCarter</title>
</head>

<body class="light vsc-initialized">
<div id="layout">
 
	<div class="row">
		<div class="col c5">
			<h2 class="aligncenter"><a href="./index.html">Calvin McCarter</a></h2>
			<p class="aligncenter slogan">AI Researcher and Engineer</p>
		</div>
		<div class="col c7 aligncenter navmenu">
			<p>
				<a href="./index.html">research</a> 
				<a href="./publications.html">publications</a> 
				<a href="./code.html" class="current">code</a>
			</p>
		</div>
	</div>
	  
	<div class="row">
		<div class="col c5 aligncenter headerimage">
			<img src="./images/profile.jpg" width="200" alt="">
		</div>
	
		<div class="col c7">
			<h2>Software</h2>
			<br><br>

			<img src="./images/kditransform-pic.jpg" width="200" class="alignleft" alt="">
			<p> The kernel density integral (KDI) transformation is a feature preprocessing method. It interpolates between the two leading feature preprocessing methods as limiting cases: linear min-max scaling and quantile transformation, offering protection from the weaknesses of each.</p>
			<p>
			<a href="https://github.com/calvinmccarter/kditransform" target="_blank">GitHub repository</a>
			<br>
			<a href="https://pypi.org/project/kditransform/" target="_blank">PyPI (pip) package</a>
			</p>

			<img src="./images/condo-pic.svg" width="200" class="alignleft" alt="">
			<p> ConDo performs domain adaptation (also known as batch correction) under confounding. Domain adaptation methods typically transform domains to match each other, but what if there are true differences between the domains that must be preserved? ConDo conditions on confounding variables while finding an optimal transformation.</p>
			<p>
			<a href="https://github.com/calvinmccarter/condo-adapter" target="_blank">GitHub repository</a>
			<br>
			<a href="https://pypi.org/project/condo/" target="_blank">PyPI (pip) package</a>
			</p>

			<img src="./images/perturbnet-logo.jpg" width="200" class="alignleft" alt="">
			<p> PerturbNet learns the gene network that modulates the influence of SNPs on phenotypes, using SNPs as naturally occurring perturbation of a biological system. PerturbNet uses a probabilistic graphical model to directly model both the cascade of perturbation from SNPs to the gene network to the phenotype network and the network at each layer of molecular and clinical phenotypes. PerturbNet learns the entire model by solving a single optimization problem with an extremely fast algorithm that can analyze human genome-wide data within a few hours.</p>

			<p>
			<a href="./software/PerturbNet.zip" target="_blank">Zipped package with CLI, Matlab, and Python interfaces</a>
			<br>
			<a href="https://github.com/SeyoungKimLab/PerturbNet" target="_blank">GitHub page for CLI, Matlab, and Python</a>
			<br>
			</p>
			<br>
			<h2>Open Source Contributions</h2>
			<br>
			<p>
			<a href="https://github.com/ToriML/onnx2pytorch" target="_blank">onnx2pytorch</a>
			<br>
			A library that converts ONNX models to PyTorch.
			<br>
			<a href="https://github.com/ToriML/onnx2pytorch/graphs/contributors" target="_blank">[main contributor]</a></a>
			</p>

			<p>
			<a href="https://github.com/mlcommons/inference" target="_blank">MLPerf Inference</a>
			<br>
			MLPerfâ„¢ Inference Benchmark Suite.
			<br>
			<a href="https://github.com/mlcommons/inference/pull/1015" target="_blank">[memory-efficient implementation of pyramidal encoder in RNN-T model]</a></a>
			</p>

			<p>
			<a href="https://github.com/tonyduan/matrix-completion" target="_blank">matrix-completion</a>
			<br>
			Python library for in-memory classical matrix completion.
			<br>
			<a href="https://github.com/tonyduan/matrix-completion/pull/3" target="_blank">[implementation of memory-efficient incremental SV thresholding]</a></a>
			</p>
			
			<p>
			<a href="https://github.com/lmcinnes/enstop" target="_blank">EnsTop</a>
			<br>
			An ensemble approach to topic modeling using pLSA.
			<br>
			<a href="https://github.com/lmcinnes/enstop/pull/8" target="_blank">[implementation of sample-weighted pLSA]</a></a>
			</p>

			<p>
			<a href="https://github.com/pytorch/pytorch" target="_blank">PyTorch</a>
			<br>
			The library that freed us from TensorFlow.
			<br>
			<a href="https://github.com/pytorch/pytorch/commits?author=calvinmccarter-at-lightmatter" target="_blank">[added lazy InstanceNorm]</a></a>
			</p>

			<p>
			<a href="https://github.com/onnx/tensorflow-onnx" target="_blank">tf2onnx</a>
			<br>
			A library that converts TensorFlow models to ONNX.
			<br>
			<a href="https://github.com/onnx/tensorflow-onnx/commits?author=calvinmccarter-at-lightmatter" target="_blank">[statically-determined output padding for ConvTranspose]</a></a>
			</p>

			<p>
			<a href="https://github.com/matsui528/nanopq" target="_blank">nanopq</a>
			<br>
			Python library for Product Quantization (PQ) and Optimized Product Quantization (OPQ).
			<br>
			<a href="https://github.com/matsui528/nanopq/graphs/contributors" target="_blank">[OPQ initialization using multivariate Gaussian assumption]</a></a>
			</p>

		</div>
	</div>
	
	<br><br><br><br><br>
 </div>


</body></html>
